crates/containerd-shim-wasm/src/sandbox/shim/otel.rs

```diff
@@ -134,31 +151,208 @@ impl Drop for ShutdownGuard {
 
 #[derive(Default)]
 pub struct ConfigBuilder {
-    otel_endpoint: Option<String>,
-    otel_protocol: Option<String>,
+    traces_endpoint: Option<String>,
+    traces_protocol: String,
 }
 
 impl ConfigBuilder {
-    /// Sets the OpenTelemetry endpoint.
-    pub fn otel_endpoint(mut self, otel_endpoint: String) -> Self {
-        self.otel_endpoint = Some(otel_endpoint);
+    /// Sets the OTLP endpoint from environment variables.
+    pub fn traces_endpoint_from_env(mut self) -> Self {
```

Comment:
**praise:** Nice! I like this, it hides the otel env-vars handling to this module :-)